packages/tools/u-boot/package.mk

```diff
@@ -18,19 +18,27 @@
 ################################################################################
 
 PKG_NAME="u-boot"
-PKG_VERSION="2017.09"
-PKG_SHA256="b2d15f2cf5f72e706025cde73d67247c6da8cd35f7e10891eefe7d9095089744"
 PKG_ARCH="arm aarch64"
 PKG_SITE="https://www.denx.de/wiki/U-Boot"
-PKG_URL="ftp://ftp.denx.de/pub/u-boot/u-boot-$PKG_VERSION.tar.bz2"
-PKG_SOURCE_DIR="u-boot-$PKG_VERSION"
+PKG_SOURCE_DIR="u-boot-$PKG_VERSION*"
```

Comment:
^^^ this won't work, as `PKG_VERSION` isn't set until later. :)

Reply:
guess it works by accident due to multiple sourcing of `package.mk`, I get `PKG_SOURCE_DIR=u-boot-5ecf0ee*` after patches are applied

Reply:
Yes, as you say it will work by chance when the package is sourced multiple times within the same process but will not work as intended when the package is sourced only once, for example:

```
$ bash
$ PROJECT=Rockchip DEVICE=RK3399 ARCH=arm source config/options u-boot; echo -e "PKG_VERSION=${PKG_VERSION}\nPKG_SOURCE_DIR=${PKG_SOURCE_DIR}"
PKG_VERSION=5ecf0ee
PKG_SOURCE_DIR=u-boot-0.0invalid*
```